### DISCUSSION TITLE
bind_rows_from in nicegui table

### DISCUSSION THREAD

**Discussion by JeanBaptiste-dlb** [zauberzeug/nicegui#3026] [Ideas / Feature Requests]
Hey, 

I started to switch from streamlit to Nicegui and it is great so far.

I would like to be able to bind the rows of a nicegui table in the following way:

```python
from pydantic import BaseModel
from typing import List
from nicegui import ui

class Document(BaseModel):
   title: str = ""

class Documents(BaseModel):
   documents: List[Document] = []

documents = Documents()

table = ui.table(
rows = [],
columns = [{"name": "title", "label": "Title", "field": "title"}]
)

table.bind_rows_from(documents, "documents", backward = lambda docs: [ {"title": doc.title} for doc in docs])

```

I believe it would be a nice feature to have. 

Thanks in advance for reading and considering this feature request

**Comment by falkoschindler** [comment]
Yes, that would be a nice feature, @JeanBaptiste-dlb!

Currently we don't provide built-in binding support for collections like lists and dictionaries, because there is no efficient way to detect changes other than repeatedly comparing the whole collection. Bindable properties like a label text or Markdown content use a special hook to detect when a property is written. But to change a list item, you don't need write access to the list reference, but only to a single item, so the list doesn't even notice the change.

Anyway, we're open for suggestions how to this feature could be implemented!
